library/core/src/main/java/com/google/android/exoplayer2/source/SimpleMediaSource.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.exoplayer2.source;
+
+import androidx.annotation.Nullable;
+import com.google.android.exoplayer2.Timeline;
+
+/**
+ * Single-child {@link CompositeMediaSource}.
+ */
+public abstract class SimpleMediaSource extends CompositeMediaSource<Void> {
+
+  /**
+   * @deprecated - Use {@link #onChildSourceInfoRefreshed}.
+   */
+  @Deprecated
+  @Override
+  protected final void onChildSourceInfoRefreshed(
+      Void id, MediaSource mediaSource, Timeline timeline
+  ) {
+    onChildSourceInfoRefreshed(mediaSource, timeline);
+  }
+
+  /**
+   * Called when the source info has been refreshed.
+   *
+   * @param mediaSource The child source whose source info has been refreshed.
+   * @param timeline The timeline of the source.
+   */
+  protected abstract void onChildSourceInfoRefreshed(
+      MediaSource mediaSource, Timeline timeline
```

Comment:
The implementation of this class knows the wrapped inner `mediaSource`, so no need to pass it back here I think.

Reply:
Not sure I agree 100% - it of course knew when it called `prepareChildSource`, but for it to know when this is called it'd have to have kept a reference to it somehow. And that is possible, but so is that a regular `CompositeMediaSource` inheritor keeps a mapping between ids and mediasources itself, so that it could retrieve the child just by using the id parameter on `CompositeMediaSource.onChildSourceInfoRefreshed`, and therefore applying this same argument then the `MediaSource` parameter wouldn't be needed there either. Without wanting to create a discussion on whether that argument should be removed in `CompositeMediaSource.onChildSourceInfoRefreshed`, I think for the sake of consistency with `CompositeMediaSource` the argument is better kept here too. Am I making sense?

Reply:
It does make sense generally, but I think all implementations of this `WrappingMediaSource` will need to have a local reference to the wrapped child media source in any case. For example, the implementation couldn't create `MediaPeriod` from `createPeriod` if it doesn't have a reference to the child source. 

> I think for the sake of consistency with CompositeMediaSource

We are already changing the method signature by removing the useless argument, so there is no consistency to start with?